easybuild/framework/extensioneasyblock.py

```diff
@@ -100,18 +100,31 @@ def __init__(self, *args, **kwargs):
         self.ext_dir = None  # dir where extension source was unpacked
 
     def _set_start_dir(self):
-        """Set value for self.start_dir
+        """Set absolute path of self.start_dir similarly to EasyBlock.guess_start_dir
 
-        Uses existing value of self.start_dir if it is already set and exists
-        otherwise self.ext_dir (path to extracted source) if that is set and exists, similar to guess_start_dir
+        Uses existing value of self.start_dir if it is already set, an absolute path and exists
+        otherwise use self.ext_dir (path to extracted source) as base dir if that is set and exists.
         """
-        possible_dirs = (self.start_dir, self.ext_dir)
-        for possible_dir in possible_dirs:
-            if possible_dir and os.path.isdir(possible_dir):
-                self.cfg['start_dir'] = possible_dir
-                self.log.debug("Using start_dir: %s", self.start_dir)
+        start_dir, ext_start_dir = '', ''
+
+        # Use provided start dir if it is an absolute path
+        if self.start_dir:
+            start_dir = self.start_dir
+            if os.path.isabs(start_dir) and os.path.isdir(start_dir):
+                self.log.info("Using user provided start dir: %s", start_dir)
+                return
```

Comment:
IIRC @boegel doesn't like using `return` to escape different code paths. Perhaps use an interim variable link `final_start_dir=None`, and then check `and not final_start_dir` below, make the assignment at the end then with an `if final_start_dir...else...`

Reply:
I do see that the use of `return` was already there 🤷 so if @boegel want's to tell me I'm mistaken...

Reply:
Yeah, not a big fan of inline `return` statements, they're easy to overlook and misinterpret what the function is doing...

Especially empty `return` statements like this, which are equivalent with `return None`...

Reply:
guys, we will have to grab a beer, sit and have a serious talk in the next EUM. Early returns are your friends :heart: 

I rewrote the code with an implicit return and simplified the logic a bit. It is indeed easier to read now.

Reply:
I feel like I should be buying that beer